draft-yun-privacypass-arc.md

```diff
@@ -411,16 +411,27 @@ obtained by deserializing a presentation according to [Section 4.3.2 of ARC],
 a presentation limit, denoted `presentation_limit`, a presentation nonce
 from a token, denoted `nonce`, and the digest of a token challenge, denoted
 `challenge_digest`, verifying a Token requires invoking the VerifyPresentation
-function from [Section 4.3.3 of ARC] in the following wayz:
+function from [Section 4.3.3 of ARC] in the following ways:
 
 ~~~
-request_context = concat(tokenChallenge.issuer_name, issuer_key_id)
-presentation_context = concat(challenge_digest, issuer_key_id)
+request_context = concat(tokenChallenge.issuer_name,
+  tokenChallenge.origin_info,
+  tokenChallenge.credential_context,
+  issuer_key_id)
+presentation_context = concat(tokenChallenge.issuer_name,
+  tokenChallenge.origin_info,
+  tokenChallenge.redemption_context,
+  issuer_key_id)
 valid = VerifyPresentation(skI, pkI, request_context, presentation_context, nonce, presentation, presentation_limit)
 ~~~
 
 This function returns True if the CredentialToken is valid, and False otherwise.
 
+Implementation-specific steps: to prevent double spending, the server should perform a check that the
+tag (presentation.tag) has not previously been seen. It then stores the tag for use in future double
```

Comment:
```suggestion
presentation (`presentation_nonce` and `presentation`) has not previously been seen for a given request and credential context. It then stores the presentation values for use in future double
```

Reply:
The server should just store the tag for double spending, as the tag is the "weight bearing" field for rate limiting purposes. The other presentation elements may be re-randomized (and therefore look different) even if the rate limit is exceeded (eg if a nonce is reused), so checking the whole presentation is not the right way to enforce rate limiting.